packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart

```diff
@@ -38,12 +38,13 @@ class TimelineScreenBody extends StatefulWidget {
 }
 
 class TimelineScreenBodyState extends State<TimelineScreenBody> {
-  @visibleForTesting
-  final controller = TimelineController();
+  TimelineController controller;
```

Comment:
This controller is only used in build, so you don't need to refer to it from the state.

Reply:
That is, this can become a stateless widget, with the build method starting with

build(...) {
  final controller = Controllers.of(context).timeline;
}

Reply:
Terry is working on a change that will be using it to subscribe to a listener, so I'll leave it as is in favor of avoiding merge conflicts.

Reply:
That is fine. Go ahead and add a TODO to document that we will be removing the stateful widget in the near future.

Reply:
This needs to be a stateful widget as is because the toggle switches between timeline modes, so we will not be removing the stateful widget. Terry is working on hooking the bar chart up to live data, for which he will be adding a listener subscription in didChangeDependencies. This is why I was saying I would just leave it because he is working on this right now.

Reply:
I'm missing where the setState call gets triggered on mode change.

Reply:
It is not part of this PR - was landed in the initial timeline flutter PR.
```
void _onTimelineModeChanged(bool frameBased) {
    setState(() {
      controller.timelineMode =
          frameBased ? TimelineMode.frameBased : TimelineMode.full;
    });
  }
```